### PR TITLE
Replaces eosjs-ecc with elliptic

### PR DESCRIPTION
**Pull request by bradlhart**
## Change Description
Updated to replace eosjs-ecc references with elliptic. Adjusted tests.
Additionally updated to bump eosjs version to edge.  Changed usages of eosjs JsonRpc.

## API Changes
- [ ] API Changes

## Documentation Additions
- [ ] Documentation Additions

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -22,7 +22,8 @@
     "bip38": "^2.0.2",
     "bs58": "^4.0.1",
     "classnames": "^2.2.6",
-    "eosjs": "^20.0.0",
+    "elliptic": "6.5.2",
+    "eosjs": "20.0.4-e2c667e.0",
```
Comment (GreenBusDriver) [inline package.json]: 2 things here
 - We don't want to be merging any of this to master.  Merging to master is only what we do once we want to release the product (which we don't yet).  Please change the base of this branch to `develop`.
 - We will *never* reference an edge release of eosjs in a `master` PR.  I think this is just a consequence of point #1, but I wanted to acknowledge the issue separately just to ensure the training...

Comment (bradlhart) [reply package.json]: Switched the PR to use develop instead of master.  Branch/commit were started from develop locally.

File: src/store/auths/authsActions.ts [hunk 2]
```diff
@@ -81,7 +83,9 @@ export const authAdd = (nickname: string, privateKey: string, passphrase: string
         throw new Error('Invalid Passphrase')
       }
 
-      if (!isValidPrivate(privateKey)) {
+      try {
+        PrivateKey.fromString(privateKey).toElliptic()
```
Comment (GreenBusDriver) [inline src/store/auths/authsActions.ts]: it'll fail on the first part; you can remove .toElliptic()

File: src/utils/Api.ts [hunk 3]
```diff
@@ -17,7 +17,7 @@ export default class Api {
   private chainId: string
 
   constructor(abis: HexAbi[], publicKeys: string[], chainId: string) {
-    this.rpc = new JsonRpc(null) // provide null chainURL to ensure vault isn't reaching out to chain for info
+    this.rpc = new JsonRpc('') // provide null chainURL to ensure vault isn't reaching out to chain for info
```
Comment (GreenBusDriver) [inline src/utils/Api.ts]: Why must this be empty string?  What was changed that required this?  Was that change really necessary?
Also, *if* this is necessary, let's move this into the function as a default arg value to make the function robust against bad input and not burden clients with having to know to add this arg to their calls.

Comment (bradlhart) [reply src/utils/Api.ts]: I've made PRs on eosjs to resolve the issue on eosjs side (two options to either allow or error)
https://github.com/EOSIO/eosjs/pull/631
https://github.com/EOSIO/eosjs/pull/632

As for this package, I believe changing it to an empty string is the better option since the JsonRpc constructor is expecting a string for endpoint with TS.  Using null instead is not good practice and is circumventing type checking.  I'm updating to change "provide null chainURL" to "provide empty chainURL" though.

File: src/store/auths/authsActions.ts (hunk 2 above)
Comment (bradlhart) [reply src/store/auths/authsActions.ts]: https://github.com/EOSIO/eosio-reference-chrome-extension-authenticator-app/pull/37/commits/d8455f0e00255a5be1781e1643d298daf1572df3#diff-b6bf3919e763d246ff62d57e2bcbc07b

File: src/utils/Api.ts (hunk 3 above)
Comment (bradlhart) [reply src/utils/Api.ts]: https://github.com/EOSIO/eosio-reference-chrome-extension-authenticator-app/pull/37/files/ead38bb2420c82514a8d719d5d2bc424a04c564c..d8455f0e00255a5be1781e1643d298daf1572df3#diff-db1f72b58945ed2936a1519c7c18f11c

File: src/utils/Api.ts [hunk 4]
```diff
@@ -17,7 +17,7 @@ export default class Api {
   private chainId: string
 
   constructor(abis: HexAbi[], publicKeys: string[], chainId: string) {
-    this.rpc = new JsonRpc(null) // provide null chainURL to ensure vault isn't reaching out to chain for info
+    this.rpc = new JsonRpc('') // provide empty chainURL to ensure vault isn't reaching out to chain for info
```
Comment (GreenBusDriver) [inline src/utils/Api.ts]: Can we just remove this value altogether?  This still makes me uncomfortable because it looks like you *have* to pass an empty string where we used to have to pass null.  If there's no value, there should just be no value, and we should default the value or deal with no value internally.  Using a value as a non-value is not high on the coolness scale... :)

File: package.json [hunk 5]
```diff
@@ -22,7 +22,8 @@
     "bip38": "^2.0.2",
     "bs58": "^4.0.1",
     "classnames": "^2.2.6",
```
Comment (GreenBusDriver) [inline package.json]: Let's lock these deps

File: src/store/auths/authsActions.ts [hunk 6]
```diff
@@ -96,7 +100,7 @@ export const authAdd = (nickname: string, privateKey: string, passphrase: string
       auths.push({
         nickname,
         encryptedPrivateKey: await encrypt(privateKey, passphrase),
-        publicKey: privateToPublic(privateKey),
```
Comment (GreenBusDriver) [inline src/store/auths/authsActions.ts:99]: BTW, one option for avoiding breaking this API is to move these toElliptic() and fromElliptic() to a separate EcConversions API, and just keep them completely separate from the primarily API.  Then the primary API remains pristine...

Comment (GreenBusDriver) [reply src/store/auths/authsActions.ts:99]: OK, we've now beaten this horse.   I agree that we should just put the PublicKey, PrivateKey, and Signature classes in the jssig file and allow people to grab them directly if they choose to.  They're just helper functions, and I think that draws that line clearly enough.

File: src/store/auths/authsActions.ts [hunk 7]
```diff
@@ -105,6 +110,19 @@ export const authAdd = (nickname: string, privateKey: string, passphrase: string
     }
   }
 
+export const recoverPublicKey = (privateKey: string, data: string) => {
```
Comment (jlamarr22) [inline src/store/auths/authsActions.ts]: This would ideally be a helper in `eosjs` or somewhere else since it's so convoluted and requires quite a bit of `elliptic` knowledge, which could be abstracted away from the consumer.

**Review by jeffreyssmith2nd — APPROVED** [review]
